kgforge/specializations/models/rdf/rdf_model_service.py

```diff
@@ -11,137 +11,47 @@
-                if predecessors:
-                    predecessors.remove(self.node)
-        else:
-            # FIXME: there are some SHACL constrains that are not implemented
-            # raise IndexError(f"{param} not implemented!")
-            pass
-
-    return properties, attributes
-
 
-class ShapeWrapper(Shape):
-    __slots__ = ('__dict__',)
-
-    def __init__(self, shape: Shape) -> None:
-        super().__init__(shape.sg, shape.node, shape._p, shape._path, shape.logger)
-
-    def parameters(self):
-        return (p for p, v in self.sg.predicate_objects(self.node)
-                if p in ALL_CONSTRAINT_PARAMETERS)
-
-
-class ShapesGraphWrapper(ShapesGraph):
-
-    def __init__(self, graph: Graph) -> None:
-        super().__init__(graph)
-        # the following line triggers the shape loading
-        self._shapes = self.shapes
-
-    def lookup_shape_from_node(self, node: URIRef) -> Shape:
-        """ Overwrite function to inject the transverse function for only to requested nodes.
-
-        Args:
-            node (URIRef): The node to look up.
-
-        Returns:
-            Shape: The Shacl shape of the requested node.
-        """
-        shape = self._node_shape_cache[node]
-        if shape:
-            shape_wrapper = ShapeWrapper(self._node_shape_cache[node])
-            if not hasattr(shape_wrapper, "traverse"):
-                shape_wrapper.traverse = types.MethodType(traverse, shape_wrapper)
-            return shape_wrapper
-        return shape
-
-
-class RdfService:
+class RdfModelService:
+    shape_to_source: Dict[URIRef, str]
+    class_to_shape: Dict[str, URIRef]
 
     def __init__(self, graph: Graph, context_iri: Optional[str] = None) -> None:
 
         if context_iri is None:
-            raise ConfigurationError(f"RdfModel requires a context")
+            raise ConfigurationError("RdfModel requires a context")
         self._graph = graph
         self._context_cache = dict()
-        self.classes_to_shapes = self._build_shapes_map()
-        resolved_context = self.resolve_context(context_iri)
-        self.context = Context(resolved_context, context_iri)
-        self.types_to_shapes: Dict = self._build_types_to_shapes()
+        self.shape_to_source, self.class_to_shape = self._build_shapes_map()
+        self.context = Context(self.resolve_context(context_iri), context_iri)
+        self.types_to_shapes: Dict[str, URIRef] = self._build_types_to_shapes()
 
-    def schema_source_id(self, schema_iri: str) -> str:
-        # POLICY Should return the id of the resource containing the schema
-        raise NotImplementedError()
+    def shape_source(self, schema_iri: URIRef) -> str:
+        return self.shape_to_source[schema_iri]
+
+    def sparql(self, query: str) -> List[Resource]:
+        e = self._graph.query(query)
+        results = json.loads(e.serialize(format="json"))
+        return SPARQLQueryBuilder.build_resource_from_select_query(results["results"]["bindings"])
```

Comment:
Should there be support for other types of queries like in store (ex: Construct queries) ?